src/main/java/bio/terra/app/controller/SearchApiController.java

```diff
@@ -83,7 +87,10 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
         @PathVariable("id") String id,
         @Valid @RequestBody SearchIndexRequest searchIndexRequest
     ) {
-        iamService.verifyAuthorization(getAuthenticatedInfo(), IamResourceType.DATASNAPSHOT, id, IamAction.READ_DATA);
+        AuthenticatedUserRequest user = getAuthenticatedInfo();
+        // Only admins have the configure action, so effectively this locks the indexing endpoint to admins only
+        iamService.verifyAuthorization(user, IamResourceType.DATAREPO, appConfig.getResourceId(), IamAction.CONFIGURE);
```

Comment:
This seems fine for now, but I think we'll need to discuss this again. As I've said before, I think using `IamResourceType.DATASET` `IamAction.LINK_SNAPSHOT` instead of `IamResourceType.DATASNAPSHOT` `IamAction.READ_DATA` would give us the correct protections to avoid letting non-admin users from creating indexes.